internal/proxy/authentication_interceptor.go

```diff
@@ -62,6 +62,9 @@ func AuthenticationInterceptor(ctx context.Context) (context.Context, error) {
 	if !ok {
 		return nil, ErrMissingMetadata()
 	}
+	if globalMetaCache == nil {
+		return nil, ErrGlobalMetaCacheUninitialized()
```

Comment:
NIT: Error message here is not user friendly. Actual case was calling a grpc before proxy is healthy.

Reply:
Yes. I will instead return errors like "proxy not initialized" or "proxy is not ready".